crates/ethereum-forks/src/forkid.rs

```diff
@@ -182,18 +182,22 @@ impl ForkFilter {
     where
         F: IntoIterator<Item = ForkFilterKey>,
     {
+        // Convert the genesis hash to a `ForkHash`.
         let genesis_fork_hash = ForkHash::from(genesis_hash);
-        let mut forks = forks.into_iter().collect::<BTreeSet<_>>();
-        forks.remove(&ForkFilterKey::Time(0));
-        forks.remove(&ForkFilterKey::Block(0));
 
+        // Process and filter forks based on criteria.
         let forks = forks
             .into_iter()
-            // filter out forks that are pre-genesis by timestamp
-            .filter(|key| match key {
-                ForkFilterKey::Block(_) => true,
-                ForkFilterKey::Time(time) => *time > genesis_timestamp,
```

Comment:
I prefer the previous `forks.remove` statements, instead of putting everything into a filter condition, it makes it more clear that forks at zero are not considered

Reply:
Sure I can revert but I still think that the new version is better in terms of performance:
- A mutable `BTreeSet` does not need to be created
- The three successive operations which require paths (the two removes then the filters) are carried out successively whereas in this new version, the complete filter is applied at once.
- The two necessary `into_iter` are therefore replaced by one since everything is applied at once upstream.

What do you think?

Reply:
the new version still collects into a `BTreeSet`, and runs `into_iter` twice, so I don't see how this is much better performance wise. This code is also applied only once at node startup, and is not performance critical, so readability is more important here imo.

Reply:
No problem, just reverted :)